include/ignition/gazebo/EntityComponentManager.hh

```diff
@@ -534,6 +534,10 @@ namespace ignition
       /// is protected to facilitate testing.
       protected: void ClearNewlyCreatedEntities();
 
+      /// \brief Clear the list of removed components so that a call to
+      /// RemoveComponent doesn't make the list grow indefinitely.
+      protected: void ClearRemovedComponents();
```

Comment:
Now that this exists, can you add a call to `ClearRemovedComponents` from https://github.com/ignitionrobotics/ign-gazebo/blob/139020c8ec0af219efd848b84b4954c2c65936e9/src/SimulationRunner.cc#L713-L716 and https://github.com/ignitionrobotics/ign-gazebo/blob/139020c8ec0af219efd848b84b4954c2c65936e9/src/gui/GuiRunner.cc#L103-L104

Reply:
942c6d4